fix(srpm): incorporate migration elements

subpop opened the pull request:
- Drop the separate compat subpackage in favor of bringing the migration components into the main rhc package.
- Replace the manual symlink with a systemd drop-in file that adds an Alias=rhcd.service to yggdrasil.service.
- Detect /etc/rhc/config.toml during upgrade and copy it to the new location.

Card ID: CCT-1127

ptoscano commented on dist/srpm/rhc.spec.in:
```diff
@@ -122,12 +116,25 @@ ln -sf ../yggdrasil/config.toml %{buildroot}%{_sysconfdir}/rhc/config.toml
 %endif
 
+%pre
+if [ $1 -eq 2 ]; then
+    if [ -f /etc/rhc/config.toml ]; then
+        cp /etc/rhc/config.toml /etc/yggdrasil/config.toml.migrated
+    fi
+fi
```
Since there is nothing that removes `/etc/rhc/config.toml`, this snippet will run not only at the first time it is upgraded, but every time. The result is that the generated `/etc/yggdrasil/config.toml` from this is generated at each upgrade, even overwriting changes made by the users.

subpop replied:
Does the file get left in place when the old rhc is removed? The RPM marks the file as `%config`; doesn't that move the file to `/etc/rhc/config.toml.rpmsave`?

subpop replied:
I think I figured it out. I forced an EL 10 system to install rhc-0.2.4 from EL9. Then I upgraded to the rhc RPM from this PR. The file `/etc/rhc/config.toml` is gone after the upgrade, and the file `/etc/yggdrasil/config.toml` contains the modified contents of the rhc config file.

I noticed during testing that uninstalling `yggdrasil` leaves the file `/etc/yggdrasil/config.toml.rpmsave` because the file was modified after the package was installed.

ptoscano commented on dist/srpm/rhc.spec.in:
```diff
@@ -122,12 +116,25 @@ ln -sf ../yggdrasil/config.toml %{buildroot}%{_sysconfdir}/rhc/config.toml
      systemctl start rhc-canonical-facts.timer
 fi
+if [ $1 -eq 2 ]; then
+    if [ -f /etc/yggdrasil/config.toml.migrated ]; then
```
Ditto.

ptoscano commented on dist/srpm/rhc.spec.in:
```diff
@@ -122,12 +116,25 @@ ln -sf ../yggdrasil/config.toml %{buildroot}%{_sysconfdir}/rhc/config.toml
 fi
+if [ $1 -eq 2 ]; then
+    if [ -f /etc/yggdrasil/config.toml.migrated ]; then
+        sed -e 's#broker#server#g' /etc/yggdrasil/config.toml.migrated > /etc/yggdrasil/config.toml
```
Hmm I think this sed expression is a bit too broad, and there's the risk of it changing also other parts than the wanted bit (imagine if the server hostname contains "broker"...), e.g. comments, etc. Maybe at least `^` to check for the beginning of the line, `=` for the key/value separator, and white spaces allowed according to what TOML supports.

ptoscano reviewed (changes requested):
In addition to the notes I left: should there be a `%bcond` (defaulting to true, perhaps) to enable or disable the compatibility bits? IMHO it may make it easier to isolate them right now, and also easier to quickly try custom builds without them.

subpop commented on dist/srpm/rhc.spec.in:
```diff
@@ -108,10 +99,13 @@ export %gomodulesmode
 install --directory %{buildroot}%{_sysconfdir}/rhc
 ln -sf yggdrasil.service        %{buildroot}%{_unitdir}/rhcd.service
-ln -sf ../yggdrasil/config.toml %{buildroot}%{_sysconfdir}/rhc/config.toml
+cat > %{buildroot}%{_unitdir}/yggdrasil.service.d/rhcd.conf << EOF
+[Install]
+Alias=rhcd.service
+EOF
```
That is a good point.

subpop commented:
I've addressed your feedback and pushed an updated commit. I added a bcond flag to the spec file as well as a meson build option. This way the conditional inclusion of the compatibility files can be mirrored from the spec file's conditional build into the underlying build system.

I also removed the %ghost entry for /etc/rhc/config.toml. The new package shouldn't keep the file around; after upgrading from 0.2.4, the file is removed (because it's owned by the rhc-0.2.4 package. I found that after upgrade, /etc/rhc/config.toml is gone and /etc/yggdrasil/config.toml has the modified contents of the old config file.
